webapp/src/components/HomePage/HomePage.tsx

```diff
@@ -1,11 +1,77 @@
 import * as React from 'react'
-import PollsPage from 'components/PollsPage'
+import { Loader } from 'decentraland-ui'
+import { Props } from './HomePage.types'
+import { t } from '@dapps/modules/translation/utils'
+import './HomePage.css'
+import { locations } from 'locations'
+import PollCards from './PollCards'
+import { PollWithAssociations } from 'modules/poll/types'
+
+const CARDS = 6
+
+export default class HomePage extends React.PureComponent<Props> {
+  componentWillMount() {
+    this.fetchDecentralandPolls()
+    this.fetchDistrictPolls()
+  }
+
+  fetchDecentralandPolls() {
+    const { onFetchPolls } = this.props
+    onFetchPolls({
+      limit: CARDS,
+      type: 'decentraland'
+    })
+  }
+
+  fetchDistrictPolls() {
+    const { onFetchPolls } = this.props
+    onFetchPolls({
+      limit: CARDS,
+      type: 'district'
+    })
+  }
+
+  handleClick = (poll: PollWithAssociations) => {
+    const { onNavigate } = this.props
+    onNavigate(locations.pollDetail(poll.id))
+  }
+
+  redirectToDecentralandPolls = () => {
+    const { onNavigate } = this.props
+    onNavigate(locations.pollsTable(1, 'decentraland', 'all'))
+  }
+
+  redirectToDistrictPolls = () => {
+    const { onNavigate } = this.props
+    onNavigate(locations.pollsTable(1, 'district', 'all'))
+  }
 
-export default class HomePage extends React.PureComponent {
   render() {
+    const { decentralandPolls, districtPolls, isLoading } = this.props
     return (
       <div className="HomePage">
-        <PollsPage />
+        {isLoading &&
+        decentralandPolls.length === 0 &&
+        districtPolls.length === 0 ? (
```

Comment:
_Super Minor_

Wouldn't this mean that if we deploy a new version or empty the database this page will load forever?